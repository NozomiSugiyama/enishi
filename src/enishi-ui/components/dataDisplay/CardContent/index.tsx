import React from "react";

import { styled } from "styled-components";

import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

const Host = styled(LinearLayout)`
    && {
        padding: 0.5rem 1rem 1rem;
    }
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout>;

export const CardContent: React.FC<Props> = ({ className, children, ...props }) => (
    <Host className={className} orientation="vertical" {...props}>
        {children}
    </Host>
);
