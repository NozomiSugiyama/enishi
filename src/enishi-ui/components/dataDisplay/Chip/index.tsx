import React from "react";

import { CircleChip } from "@/enishi-ui/components/dataDisplay/Chip/CircleChip";
import { NormalChip } from "@/enishi-ui/components/dataDisplay/Chip/NormalChip";

type Props =
    | (React.ComponentPropsWithRef<typeof CircleChip> & { circle: true })
    | (React.ComponentPropsWithRef<typeof NormalChip> & { circle?: false });

export const Chip: React.FC<Props> = (props) => {
    // eslint-disable-next-line react/destructuring-assignment
    if (props.circle) {
        const { circle: _, ...circleChipProps } = props;
        return <CircleChip {...circleChipProps} />;
    }

    const { circle: _, ...defaultChipProps } = props;
    return <NormalChip {...defaultChipProps} />;
};
