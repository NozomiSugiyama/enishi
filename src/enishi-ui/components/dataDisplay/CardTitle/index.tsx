import React from "react";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";

type Props = React.ComponentPropsWithRef<typeof Typography> & {
    readonly children?: React.ReactNode;
};

export const CardTitle: React.FC<Props> = ({ children, ...props }) => (
    <Typography colorTheme="secondary" {...props}>
        {children}
    </Typography>
);
