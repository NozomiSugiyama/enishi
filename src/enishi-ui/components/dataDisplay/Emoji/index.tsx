import React from "react";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";

type Props = React.ComponentPropsWithRef<typeof Typography> & {
    readonly emoji: string;
};

export const Emoji: React.FC<Props> = ({ emoji, children, ...props }) => (
    <Typography {...props}>
        <span role="img">{emoji}</span>
        {children}
    </Typography>
);
