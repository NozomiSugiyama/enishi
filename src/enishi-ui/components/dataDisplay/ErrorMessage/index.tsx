import React from "react";

import { MdError } from "@react-icons/all-files/md/MdError";
import { styled, useTheme } from "styled-components";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";
import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

const Host = styled(LinearLayout)`
    && {
        margin-left: 0.2rem;
        margin-top: 0.1rem;
        > :first-child {
            margin-right: 0.2rem;
        }
    }
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout>;

export const ErrorMessage: React.FC<Props> = ({ children, ...props }) => {
    const theme = useTheme();
    return (
        <Host gravity="center_left" {...props}>
            <MdError color={theme.palette.error[100]} />
            <Typography color={theme.palette.error[100]} data-w-id="form-item-container-error-message-field" variant="caption">
                {children}
            </Typography>
        </Host>
    );
};
