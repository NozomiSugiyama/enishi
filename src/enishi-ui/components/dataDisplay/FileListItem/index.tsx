import React from "react";

import { FiFileText } from "@react-icons/all-files/fi/FiFileText";

import { ListItem } from "@/enishi-ui/components/dataDisplay/ListItem";
import { ListItemIcon } from "@/enishi-ui/components/dataDisplay/ListItemIcon";

export const FileListItem: React.FC<React.ComponentPropsWithRef<typeof ListItem>> = ({ children, ...props }) => (
    <ListItem {...props}>
        <ListItemIcon size={props.size}>
            <FiFileText />
        </ListItemIcon>
        {children}
    </ListItem>
);
