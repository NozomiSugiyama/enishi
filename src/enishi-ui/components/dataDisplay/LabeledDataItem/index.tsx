import React from "react";

import { styled } from "styled-components";

import { Typography } from "@/enishi-ui/components/dataDisplay/Typography";
import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

const sizes = {
    "x-small": {
        host: {
            padding: "0.3rem 0",
            fontSize: "0.8rem"
        },
        label: {
            marginBottom: "0.1rem",
            fontSize: "0.7rem"
        },
        iconSize: "1rem"
    },
    "small": {
        host: {
            padding: "0.4rem 0",
            fontSize: "0.9rem"
        },
        label: {
            marginBottom: "0.1rem",
            fontSize: "0.8rem"
        },
        iconSize: "1rem"
    },
    "medium": {
        host: {
            padding: "0.5rem 0",
            fontSize: "1rem"
        },
        label: {
            marginBottom: "0.4rem",
            fontSize: "0.8rem"
        },
        iconSize: "1rem"
    },
    "large": {
        host: {
            padding: "0.6rem 0",
            fontSize: "1.2rem"
        },
        label: {
            marginBottom: "0.4rem",
            fontSize: "0.8rem"
        },
        iconSize: "1rem"
    },
    "x-large": {
        host: {
            padding: "0.7rem 0",
            fontSize: "1.6rem"
        },
        label: {
            marginBottom: "0.4rem",
            fontSize: "0.9rem"
        },
        iconSize: "1rem"
    }
};

const Host = styled(LinearLayout)<{
    $size: keyof typeof sizes;
}>`
    && {
        padding: ${({ $size }) => sizes[$size].host.padding};
        font-size: ${({ $size }) => sizes[$size].host.fontSize};
    }
`;

const Header = styled(LinearLayout).attrs({
    orientation: "horizontal"
})<{
    $size: keyof typeof sizes;
}>`
    margin-bottom: ${({ $size }) => sizes[$size].label.marginBottom};
`;

const Label = styled(Typography)<{
    $size: keyof typeof sizes;
}>`
    && {
        white-space: nowrap;
        font-size: ${({ $size }) => sizes[$size].label.fontSize};
    }
`;

const IconContainer = styled.div<{ $size: keyof typeof sizes }>`
    display: flex;
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout> & {
    readonly label?: string;
    readonly labelProps?: React.ComponentPropsWithRef<typeof Typography>;
    readonly size?: keyof typeof sizes;
    readonly fontWeight?: React.ComponentPropsWithRef<typeof Typography>["fontWeight"];
    readonly labelButton?: React.ReactNode | null;
};

export const LabeledDataItem: React.FC<Props> = ({ label, children, labelProps, size = "medium", fontWeight, labelButton, ...props }) => (
    <Host $size={size} orientation="vertical" {...props}>
        <Header $size={size} gravity="center">
            <Label $size={size} colorTheme="secondary" fontWeight={fontWeight} variant="caption" {...labelProps}>
                {label}
            </Label>
            {labelButton ? <IconContainer $size={size}>{labelButton}</IconContainer> : null}
        </Header>
        {children}
    </Host>
);
