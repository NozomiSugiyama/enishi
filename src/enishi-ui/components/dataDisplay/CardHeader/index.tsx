import React from "react";

import { styled } from "styled-components";

import { CardTitle } from "@/enishi-ui/components/dataDisplay/CardTitle";
import { LinearLayout } from "@/enishi-ui/components/layout/LinearLayout";

type Size = "large" | "medium";
const Host = styled(LinearLayout)<{ size: Size }>`
    padding: ${({ size }) => (size === "medium" ? "0.5rem 1rem 0.3rem" : "1rem 1.5rem 0.5rem")};
`;

type Props = React.ComponentPropsWithRef<typeof LinearLayout> & {
    readonly size?: Size;
    readonly onClose?: () => void;
};

export const CardHeader: React.FC<Props> = ({ size = "medium", children, ...props }) => (
    <Host gravity="center_left" orientation="horizontal" size={size} {...props}>
        {typeof children === "string" ? <CardTitle variant={size === "large" ? "h4" : "h6"}>{children}</CardTitle> : children}
    </Host>
);
