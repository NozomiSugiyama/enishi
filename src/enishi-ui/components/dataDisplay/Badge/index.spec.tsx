import React from "react";

import { describe, expect, it } from "@jest/globals";
import "jest-styled-components";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { Badge } from "@/enishi-ui/components/dataDisplay/Badge";

import { createThemes } from "@/theme";

const { theme } = createThemes("#1F4682");

describe("<Badge />", () => {
    it("should merge user classes & spread custom props to the root node", () => {
        const component = renderer.create(
            <ThemeProvider theme={theme}>
                <Badge className="my-badge" data-my-prop="notificationBadge" />
            </ThemeProvider>
        );
        let tree = component.toJSON() as renderer.ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();

        void renderer.act(() => {
            tree.props as React.ComponentPropsWithRef<typeof Badge>;
        });
        tree = component.toJSON() as renderer.ReactTestRendererJSON;
        expect(tree).toMatchSnapshot();
    });
});
