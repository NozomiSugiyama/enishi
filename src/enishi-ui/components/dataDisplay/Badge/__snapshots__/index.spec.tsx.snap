// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should merge user classes & spread custom props to the root node 1`] = `
.c0 {
  position: absolute;
  top: initial;
  bottom: initial;
  left: initial;
  right: initial;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #1F4682;
  border-radius: 30px;
  color: #E9EDF3;
  font-size: 0.8rem;
  font-weight: bold;
  padding: 0rem 0.5rem 0.1rem 0.5rem;
  box-sizing: border-box;
  text-align: center;
}

.c0.c0 {
  border-radius: 50%;
  min-width: 0.6rem;
  width: 0.6rem;
  max-width: 0.6rem;
  min-height: 0.6rem;
  height: 0.6rem;
  max-height: 0.6rem;
  padding: 0;
}

<div
  className="c0 my-badge"
  data-my-prop="notificationBadge"
/>
`;

exports[`<Badge /> should merge user classes & spread custom props to the root node 2`] = `
.c0 {
  position: absolute;
  top: initial;
  bottom: initial;
  left: initial;
  right: initial;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #1F4682;
  border-radius: 30px;
  color: #E9EDF3;
  font-size: 0.8rem;
  font-weight: bold;
  padding: 0rem 0.5rem 0.1rem 0.5rem;
  box-sizing: border-box;
  text-align: center;
}

.c0.c0 {
  border-radius: 50%;
  min-width: 0.6rem;
  width: 0.6rem;
  max-width: 0.6rem;
  min-height: 0.6rem;
  height: 0.6rem;
  max-height: 0.6rem;
  padding: 0;
}

<div
  className="c0 my-badge"
  data-my-prop="notificationBadge"
/>
`;
